dashboard/main.py

```diff
@@ -0,0 +1,165 @@
+"""Script for Streamlit dashboard."""
+
+import pandas as pd
+import altair as alt
+import streamlit as st
+from datetime import timedelta
+
+
+RESAMPLE_RATE = '10T'
+LINE_WIDTH = 2
+FIG_WIDTH = 1200
+FIG_HEIGHT = int(FIG_WIDTH / 4)
+
+st.set_page_config(layout="wide")
+
+
+def get_moisture_data(data: pd.DataFrame) -> pd.DataFrame:
+    """Gets moisture data from filtered dataframe."""
+
+    return data.resample(
+        RESAMPLE_RATE, on='recording_taken').mean().reset_index()
+
+
+def get_soil_moisture_chart(chart_data: pd.DataFrame) -> alt.Chart:
+
+    y_min = chart_data['soil_moisture'].min()
+    y_max = chart_data['soil_moisture'].max()
+
+    chart = alt.Chart(chart_data).mark_line(
+        interpolate='cardinal',
+        tension=0.8,
+        strokeWidth=LINE_WIDTH,
+    ).encode(
+        x=alt.X('recording_taken:T', axis=alt.Axis(
+            title='Time', format='%H:%M')),
+        y=alt.Y('soil_moisture:Q', scale=alt.Scale(
+            domain=[y_min - 1, y_max + 1]), axis=alt.Axis(title='Soil Moisture (%)'))
+    ).properties(
+        width=FIG_WIDTH,
+        height=FIG_HEIGHT,
+    ).configure_axis(
+        # labelFontSize=TICK_LABEL_FONT_SIZE,
+        # titleFontSize=AXIS_LABEL_FONT_SIZE,
+    ).configure_title(
+        # fontSize=TITLE_FONT_SIZE,
+    )
+
+    return chart
+
+
+def get_temperature_data(data: pd.DataFrame) -> pd.DataFrame:
+    """Gets moisture data from filtered dataframe."""
+
+    return data.resample(
+        RESAMPLE_RATE, on='recording_taken').mean().reset_index()
+
+
+def get_temperature_chart(chart_data: pd.DataFrame) -> alt.Chart:
+
+    y_min = temperature_data['temperature'].min()
+    y_max = temperature_data['temperature'].max()
+
+    chart = alt.Chart(temperature_data).mark_line(
+        interpolate='cardinal',
+        tension=0.8,
+        strokeWidth=LINE_WIDTH,
+    ).encode(
+        x=alt.X('recording_taken:T', axis=alt.Axis(
+            title='Time', format='%H:%M')),
+        y=alt.Y('temperature:Q', scale=alt.Scale(
+            domain=[y_min - 1, y_max + 1]), axis=alt.Axis(title='Temperature (°C)'))
+    ).properties(
+        width=FIG_WIDTH,
+        height=FIG_HEIGHT,
+    ).configure_axis(
+        # labelFontSize=TICK_LABEL_FONT_SIZE,
+        # titleFontSize=AXIS_LABEL_FONT_SIZE,
+    ).configure_title(
+        # fontSize=TITLE_FONT_SIZE,
+    )
+
+    return chart
+
+
+if __name__ == "__main__":
+
```

Comment:
As you have already said, can definitely split this into functions

Reply:
will do!